types/type-specs.tsx

```diff
@@ -66,14 +63,36 @@ const invalidParamsWithGeneric: Params<{ id: number }> = { id: 13 }; // $ExpectE
   {(params: Params): React.ReactNode => `User id: ${params.id}`}
 </Route>;
 
-<Route<{ id: string }> path="/users/:id">{({ id }) => `User id: ${id}`}</Route>;
+<Route path="/users/:id">{({ id }) => `User id: ${id}`}</Route>;
 
-<Route<{ id: string }> path="/users/:id">
+<Route path="/users/:id">
   {({ age }) => `User age: ${age}`} // $ExpectError
 </Route>;
 
+<Route path="/users/:id">
+  {({ age }: { age: string }) => `User age: ${age}`}
+</Route>;
+
 <Route path="/app" match={true} />; // $ExpectError
 
+// inferred rest params
+<Route path="/path/:rest*">{(params) => `Rest: ${params.rest}`}</Route>;
+
+// infer multiple params
+<Route path="/path/:first/:second/another/:third">
+  {({ first, second, third }) => `${first}, ${second}, ${third}`}
+</Route>;
+
+// infer only named params
+<Route path="/:first/:second">
+  {({ first, second }) => `first: ${first}, second: ${second}`}
+</Route>;
+
+// for pathToRegexp matcher
+<Route path="/:user([a-z]i+)/profile/:tab/:first+/:second*">
+  {({ user, tab, first, second }) => `${user}, ${tab}, ${first}, ${second}`}
```

Comment:
@itsMapleLeaf  Trying the suggestion feature on this one as I don't see the changes.
Thanks, dude, didn't know about that feature!

```suggestion
  {({ user, tab, first, second }) => {
    user; // $ExpectType string
    tab; // $ExpectType string
    first; // $ExpectType string
    second; // $ExpectType string | undefined
    return `${user}, ${tab}, ${first}, ${second}`;
  }}
```

Reply:
Ah, I didn't push 😂 This works!